README.md

```diff
@@ -6,52 +6,64 @@ Try the current version at: [https://cs121-team-panda.github.io/coconut-interpre
 
-## Installation
-```
-pip install -r requirements.txt
-```
+## Architecture 
+<img width="1093" alt="architecture" src="https://user-images.githubusercontent.com/35832643/38783767-b5ef5236-40bb-11e8-91b4-e1d5bdc0aa18.png">
 
-## Running the app
+The Coconut Online Interpreter project's architecture design is based on server-less computing based on AWS Lambda for executing user-submitted Coconut codes. The front-end of the project is developed using React. Based on the designed architecture shown above, the interpreter is resilient to any malicious code submitted by users since the backend protection is handled by AWS Lambda services.
+
+### Prerequisites
 Requires Python 3. 
 ```
 python manage.py runserver 
 ```
 
-# Features
+## Installation
+```
+pip install -r requirements.txt
+```
+
+## Functionality
+
+The interpreter consists of an easy-to-use webpage (shown below) that enables user to easily enter and execute Coconut code. 
 
-The interpreter consists of a simple, elegant webpage that enables users to easily enter and execute Coconut code.
+![demo1](https://user-images.githubusercontent.com/35832643/39082318-1b725362-4506-11e8-8403-7653622c40ae.gif)
 
-![ui](https://user-images.githubusercontent.com/35832643/37873776-fcecfff2-2fd7-11e8-991b-339363e8489e.png)
+**Compilation Settings:** Clicking the settings gear will display a settings drawer. From within the drawer, the user can select which version of Python the Coconut code will compile to. By default, the latest version of Python 3 is selected. 
 
-On the right, the user can see loading dots as the code loads in the output section:
+**Saving to a Coconut File:** Clicking the save icon will download the contents of the code editor into a Coconut file called ```coconut.coco```.
 
-![dots](https://user-images.githubusercontent.com/35832643/37873777-0d66bf4e-2fd8-11e8-8f14-555df97d3312.png)
+**Code Execution:** Clicking `RUN`or using the keyboard shortcut `Ctrl + Enter` for Windows or `⌘ + Enter` for Mac will compile and execute the program.
 
-On the left, we can use the code editor. The editor allows for the keyboard shortcuts from Ace editor (listed [here](https://github.com/ajaxorg/ace/wiki/Default-Keyboard-Shortcuts)). In addition, the editor includes Python and Coconut syntax highlighting for all keywords (Coconut specific keywords listed [here](http://coconut.readthedocs.io/en/latest/DOCS.html#keywords)):
+**Showing the Compiled Python:** Checking the box labeled `PYTHON` will replace the output with the compiled Python, which is the output of the Coconut compiler. The Python code will be displayed using Python syntax highlighting.
 
-![coco specific](https://user-images.githubusercontent.com/35832643/37873781-17561450-2fd8-11e8-8659-1c70463a48aa.png)
+**Code Editor:** The code editor, which is on the left side of the webpage, is where the user types in their Coconut code. The editor has line numbers and Coconut-specific syntax highlighting.
```

Comment:
Consider putting Code Editor and Code Execution Output first, before Compilation Settings, etc.